src/main/java/org/llorllale/cactoos/matchers/IsMatcherOf.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) for portions of project cactoos-matchers are held by
+ * Yegor Bugayenko, 2017-2018, as part of project cactoos.
+ * All other copyright for project cactoos-matchers are held by
+ * George Aristy, 2018.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.llorllale.cactoos.matchers;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.StringDescription;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+/**
+ * Matcher to test {@link org.hamcrest.Matcher} objects.
+ *
+ * <p>Here is a few examples how we are testing matchers
+ *  without/with {@link IsMatcherOf}:</p>
+ * <pre>{@code
+ *  @Test
+ *  public void matchesSafelyWithout() {
+ *      final Description expected = new StringDescription();
+ *      final Description actual = new StringDescription();
+ *      final HasValues<Integer> hvalues = new HasValues<>(5);
+ *      hvalues.describeTo(expected);
+ *      MatcherAssert.assertThat(
+ *          "The testing status is 'false'",
+ *          hvalues.matchesSafely(new ListOf<>(1, 2, 3), actual),
+ *          new IsEqual<>(false)
+ *      );
+ *      MatcherAssert.assertThat(
+ *          "The actual result message in hamcrest terms is correct",
+ *          actual.toString(),
+ *          new IsEqual<>("The function applied to <[1, 2, 3]> is failed")
+ *      );
+ *  }
+ *
+ *  @Test
+ *  public void matchesSafelyWith() {
+ *      MatcherAssert.assertThat(
+ *          new HasValues<>(5),
+ *          new IsMatcherOf<>(
+ *              false, new ListOf<>(1, 2, 3),
+ *               "The function applied to <[1, 2, 3]> is failed"
+ *          )
+ *       );
+ *   }
+ *
+ * }</pre>
+ *
+ * @param <X> Type of item.
+ * @since 1.0.0
+ * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
+ */
+public final class IsMatcherOf<X> extends
```

Comment:
@dgroup I'm thinking `MatcherOf` would be a slightly better name. That way it composes nicely with `IsNot`. WDYT?

Reply:
@llorllale , we already have a matcher with this name - org.llorllale.cactoos.matchers.MatcherOf
```java
/**
 * Func as Matcher.
 *
 * <p>There is no thread-safety guarantee.
 *
 * @param <T> Type of object to match
 * @since 0.12
 */
```
In case if we want to use this name the new package is required.

Reply:
@dgroup I had forgotten about this matcher.... 

The existing `MatcherOf` should instead have been a `MatcherEnvelope`, I think.

Let's use the name `Matches` for now.

Reply:
@llorllale, done.